src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -93,6 +96,12 @@ message InvocationStateResponse {
     }
 }
 
+message ListTasksRequest {}
+
+message ListTasksResponse {
+  repeated RunTaskResponse tasks = 1;
```

Comment:
Shouldn't this list all tasks including these that didn't finish? Also I'd prefer if there was a single way to obtain task outputs/results like with the await_task RPC.

Reply:
Rebuilt. (The goal was to evolve the code to where we wanted it to be instead of doing everything all at once, but I'm fine doing everything all at once :-)